services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/handlers/RemoveBucketFromPathInterceptor.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3.internal.handlers;
+
+import software.amazon.awssdk.core.interceptor.Context;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.interceptor.ExecutionInterceptor;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.utils.StringUtils;
+import software.amazon.awssdk.utils.http.SdkHttpUtils;
+
+public class RemoveBucketFromPathInterceptor implements ExecutionInterceptor {
```

Comment:
I don't know why this interceptor is needed, can we add a few lines of documentation explaining its role?

Reply:
I'm just going to remove this for now. In E2.0, the S3 rules handle setting the bucket in the URI (either on the path or in the host) so we need to actually remove the one that the marshaller sets. But instead of doing this way which is actually incorrect, we will preprocess the model at codegen time instead.